torch/_refs/__init__.py

```diff
@@ -929,6 +933,37 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _xlogy(a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]):
+    assert isinstance(a, TensorLike) or isinstance(b, TensorLike)
+
+    # 1) Operations derived from elementwise_unary_scalar_wrapper contain non-ref
+    # functions, which causes a runtime error in TorchRefsMode context manager
+    # when strict = True
+    # 2) where requires all tensors to be on the same device
+    # TODO remove scalar_tensor
+    if isinstance(a, TensorLike):
+        if isinstance(b, Number):
+            b = scalar_tensor(b, dtype=a.dtype, device=a.device)
+        elif utils.is_cpu_scalar_tensor(b):
+            b = prims.device_put(b, device=a.device)
+    elif isinstance(b, TensorLike):
+        if isinstance(a, Number):
+            a = scalar_tensor(a, dtype=b.dtype, device=b.device)
+        elif utils.is_cpu_scalar_tensor(a):
+            a = prims.device_put(a, device=b.device)
```

Comment:
What happens if the scalar tensors aren't put on the same device? Should we instead fix something in where?

Reply:
I placed the scalar tensors on the same device as the TensorLike operand.
aten where also expects all arguments to be on the same device.

Reply:
Right -- although that's a bug in torch.where, and we might want to think about fixing it in our refs

Reply:
From offline discussion: let's just look at fixing `torch.where`

Reply:
@IvanYashchuk: probably just need to correctly disable same device check in native_functions.yaml

Reply:
Yes we've agreed that `where` should start accepting cpu scalar args, but the fix is a bit more involved than just disabling same device check in native_functions.yaml.